backend/src/routes/api/status/statusUtils.ts

```diff
@@ -1,44 +1,36 @@
 import { FastifyRequest } from 'fastify';
 import { KubeFastifyInstance, KubeStatus } from '../../../types';
-import { CoreV1Api, CustomObjectsApi } from '@kubernetes/client-node';
+import { CustomObjectsApi } from '@kubernetes/client-node';
 import { getUserName } from '../../../utils/userUtils';
 import { createCustomError } from '../../../utils/requestUtils';
+import { getDashboardConfig } from '../../../utils/resourceUtils';
 
 type groupObjResponse = {
   users: string[] | null;
 };
 
-const SYSTEM_AUTHENTICATED = 'system:authenticated';
-const GROUPS_CONFIGMAP_NAME = 'groups-config';
+//const SYSTEM_AUTHENTICATED = 'system:authenticated';
 
-const getAdminUsers = async (fastify: KubeFastifyInstance): Promise<string[]> => {
-  let adminUsers: string[] = [];
+const ADMIN_GROUP = 'adminGroups';
+const ALLOWED_GROUP = 'allowedGroups';
 
-  try {
-    const groupConfig = await getGroupsConfig(fastify.kube.coreV1Api, fastify.kube.namespace);
-    const adminGroup = await getGroupsAdminConfig(
-      fastify.kube.coreV1Api,
-      fastify.kube.namespace,
-      groupConfig,
-    );
+const getUserList = async (fastify: KubeFastifyInstance, groupName: string): Promise<string[]> => {
+  let userList: string[] = [];
 
-    if (adminGroup === SYSTEM_AUTHENTICATED || adminGroup === '') {
-      throw new Error(
-        'It is not allowed to set system:authenticated or an empty string as admin group.',
-      );
-    } else {
-      adminUsers = await getGroup(fastify.kube.customObjectsApi, adminGroup);
-    }
+  try {
+    const dashCR = getDashboardConfig().spec;
+    const userGroup = dashCR.groupsConfig[groupName];
```

Comment:
`groupConfig` is not required. You need to condition this otherwise when we don't have a full CR you'll hit an NPE.

```suggestion
    const userGroup = dashCR.groupsConfig?.[groupName];
```